fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintainer.java

```diff
@@ -154,42 +141,91 @@ private void parseFieldNames(KeyExpression expression, @Nonnull String parentFie
+                                prefix = prefix == null ? subPrefix : prefix.concat("_").concat(subPrefix);
+                                List<LuceneFieldKeyExpression> children = ((LuceneThenKeyExpression)expression).getLuceneChildren();
+                                for (int j = 0; j < children.size(); j++) {
+                                    if (j != prefixLocation) {
+                                        LuceneFieldKeyExpression child = children.get(j);
+                                        Object value = entryKey.get(i + offset + j);
+                                        if (value != null) {
+                                            insertDocumentField(child, entryKey.get(i + offset + j), document, prefix);
+                                            fullText.add(value.toString());
+                                        }
+                                    }
+                                }
+                                offset += children.size();
+                            } else if (expression instanceof LuceneFieldKeyExpression) {
+                                Object value = entryKey.get(i + offset);
+                                if (value != null) {
+                                    insertDocumentField((LuceneFieldKeyExpression)expression, value, document, null);
+                                    fullText.add(value.toString());
+                                }
                             }
-                            document.add(new TextField(fieldNames.get(i), value, Field.Store.NO));
                         }
-                        writer.addDocument(document);
                     }
                 } catch (IOException ioe) {
                     throw new RecordCoreException("Issue updating index keys", "Key", entry, "IndexEntries", indexEntries,
                             "savedRecord", savedRecord, ioe);
                 }
             });
+            String fullTextString = "";
+            for (String value : fullText) {
+                fullTextString = fullTextString.concat(" ").concat(value);
+            }
+            document.add(new TextField(FULL_TEXT_KEY_FIELD, fullTextString, Field.Store.NO));
+            writer.addDocument(document);
         } catch (Exception e) {
             throw new RecordCoreException(e);
         }
         return AsyncUtil.DONE;
     }
 
+    private void insertDocumentField(final LuceneFieldKeyExpression expression, Object value,
+                                     final Document document, String prefix) {
+        if (!(value == null && expression.getType() != LuceneKeyExpression.FieldType.STRING)) {
+            String fieldName = prefix == null ? expression.getFieldName() : prefix.concat("_").concat(expression.getFieldName());
+            switch (expression.getType()) {
+                case INT:
+                    document.add(new IntPoint(fieldName, (Integer)value));
```

Comment:
Should be possible to have a stored int field, I think. Maybe just a `TODO` for now.

Reply:
It should, We need to think how to handle this since IntPoint is inherently not a stored field. Its also possible to have a sorted numeric field, there are a few ways of handling numeric values in Lucene depending on what you want them for.